packages/@sanity/types/src/schema/types.ts

```diff
@@ -31,6 +33,15 @@ export type SortOrdering = {
   }
 }
 
+export interface HiddenPredicateContext {
+  parent?: Record<string, unknown>
```

Comment:
I think parent can be an array as well

Reply:
Currently only objects are supported. Conditionally hiding array items calls for a different set of considerations and is out of scope for now.

Reply:
oh yeah you're right. i don't really think it makes sense to hide an item of an array ever 😄